File: BaseLib/src/main/java/com/hjc/baselib/dialog/LoadingDialog.java
package com.hjc.baselib.dialog;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;

import com.hjc.baselib.R;
import com.hjc.baselib.databinding.DialogLoadingBinding;
import com.hjc.baselib.viewmodel.BaseViewModel;

/**
 * @Author: HJC
 * @Date: 2019/1/7 11:28
 * @Description: 加载框
 */
public class LoadingDialog extends BaseFragmentDialog<DialogLoadingBinding, BaseViewModel> {

    public static LoadingDialog newInstance() {
        return new LoadingDialog();
    }

    @Override
    public int getStyleRes() {
        return R.style.Dialog_Base;
    }

    @Override
    public int getLayoutId() {
        return R.layout.dialog_loading;
    }

    @Override
    protected BaseViewModel getViewModel() {
        return null;
    }

    @Override
    protected int getBindingVariable() {
        return 0;
    }

    @Override
    public void initData(@Nullable Bundle savedInstanceState) {
        //去掉遮盖层
//        Window window = getDialog().getWindow();
//        if (window != null) {
//            window.setDimAmount(0f);
//        }
        setCancelable(true);
    }

//    @SuppressLint("CheckResult")
//    public void dismissDialog(){
//        Observable.timer(500, TimeUnit.MILLISECONDS)
//                .compose(RxSchedulers.ioToMain())
//                .subscribe(aLong -> dismissAllowingStateLoss());
//    }


    @Override
    public void addListeners() {

    }

    @Override
    public void onSingleClick(View v) {

    }
}

File: BaseLib/src/main/java/com/hjc/baselib/dialog/LoadingDialog.kt
package com.hjc.baselib.dialog

import android.os.Bundle
import android.view.View
import com.hjc.baselib.R
import com.hjc.baselib.databinding.DialogLoadingBinding
import com.hjc.baselib.viewmodel.CommonViewModel

/**
 * @Author: HJC
 * @Date: 2019/1/7 11:28
 * @Description: 加载框
 */
class LoadingDialog : BaseFragmentDialog<DialogLoadingBinding, CommonViewModel>() {

    companion object {
        fun newInstance(): LoadingDialog {
            return LoadingDialog()
        }
    }

    override fun getStyleRes(): Int {
        return R.style.Dialog_Base
    }

    override fun getLayoutId(): Int {
        return R.layout.dialog_loading
    }

    override fun createViewModel(): CommonViewModel? {
        return null
    }

    override fun initData(savedInstanceState: Bundle?) {
        //去掉遮盖层
        dialog?.let {
            val window = it.window
            window?.setDimAmount(0f)
        }
        isCancelable = true
    }

    override fun addListeners() {}

    override fun onSingleClick(v: View?) {}

}